docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,237 @@
+
+### MJML
+
+As part of modernizing the templates, we aim to replace the framework in which our templates are currently built in so as to include more logic in them while development. Currently, we've decided on using [MJML](https://mjml.io/) as our potential markup along with a templating framework like React, Mustache, or EJS. However, there needs to be more discussion around it so the following list of pros and cons could be pondered upon to decide whether or not we should use MJML as part of our templating stack.
+
+- Pros
+  - MJML has its own custom tags defined which translates to responsive HTML after compilation
+  - MJML can help us create responsive emails using mobile-first approach
+  - It is a component-based library making the components reusable across the templates
+  - This library has been created by keeping in mind all the industry standards as well as the quirks of old email clients, so we will eventually be saving time on investigating cross-browser compatability of our templates
+  - MJML has pre-built standard components as well as custom components which could be modified by passing parameters. E.g. this social links component
+  - MJML has by far the highest weekly downloads and github stars in comparision to its alternatives like Foundation for Emails
+- Cons
+  - Although MJML is a responsive library, it controls that responsiveness automatically needing very less to no developer intervention
+
+### CSS Styles
+
+Chosen option: [todo], because:
+
+- [todo]
+
+### Email Previewing
+
+Chosen option: "Option B - Use Storybook", because:
+
+- It is easy to setup and configure.
+- We have a Storybook deployment already in place and has worked great so far for a few other FxA services so hooking up the client for `fxa-auth-server` to view the modernized components might result in consistency.
+- Easy to test and perform QA. The design and product teams don't have to touch the codebase to test the various states of UI.
+- Support for CSS (whether it's plain old CSS, TailwindCSS or something else)
+
+### Other
+
+**Plaintext files**: [todo]
+
+**Localization**: We will upgrade the localization tool from GetText to Fluent since it's preferred by the l10n team and other FxA front-ends are using it.
+
+**Integration & QA**: [todo]
+We also intend on making a plan on how we're going to build out/hook refactored templates up with feature flags and mitigate risk. The `fxa-auth-server` already has some basic support for feature flagging, but this ADR aims to identify how we want to use these - do we want 3 different feature flags (one for each milestone of templates), do we want to flag individual templates, or should we have a single flag?
```

Comment:
(note to anyone reading this: @jodyheavener, @vbudhram, and I are going to write out a section here)

Reply:
@LZoog @Yogita98 How about something like this:

> During development templates will be marked as being part of a release group. This could be in the form of a mapped list of template names or some variable associated with the template file. Each group will have a corresponding environment variable flag. When this release group’s flag is enabled any templates that fall under it will be used when generating an email template in that environment; until a template's corresponding release group is enabled it is not used and the old/current template will continue to be served. Release groups will be initially enabled in staging until QA has had an opportunity to thoroughly test and evaluate each template in the group, after which they can be incrementally enabled in production.

Reply:
It looks good to me. I'll include it in the ADR